modules/stochastic_tools/include/utils/MultiDimPolynomialGenerator.h

```diff
@@ -0,0 +1,48 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "libmesh/utility.h"
+
+#include "DataIO.h"
+#include "CartesianProduct.h"
+
+/**
+ * Polynomials and quadratures based on defined distributions for Polynomial Chaos
+ */
+namespace StochasticTools
+{
+/**
+ Class containing functionalitties to generate the degress of the 1D Polynomials
+ within every term of a multi-dimensional polynomial.
+ */
+class MultiDimPolynomialGenerator
```

Comment:
You don't need a class here, since the two methods are both static. Just make a nested namespace, the use will be the same.

Reply:
So I was thinking a bit about this class when I wrote the code and I think it might be nice to implement a generic class for polynomials (Hermite, Lagrange, Regular etc) in the utilities. Just like the one which @zachmprince uses for the quadratures in PCE but without the quadrature options. I think then this multiD polynomial class could inherit from the 1D polynomial class and create multiD polynomials. I think in that class these functions wouldn't be static. This is the reason why I created a class instead of a namespace.  Let me know what you think. Should I still change it to a nested namespace?

Reply:
This is actually a decent idea from Peter. But I don't think this level of abstraction is needed right now. If we find another application that needs these polynomials or polynomial regression would like to use orthogonal polynomials, then we can work on making multiD polynomials more general. For now, lets just go with the namespace implementation.

Reply:
I'll buy it.